### DISCUSSION TITLE
SIPDG Poisson example

### DISCUSSION THREAD

**Discussion by fdrmrc** [fdrmrc/non_matching_test_suite#41] [Show and tell]
After #27 and #26, the SIP applied to the Poisson example has the following assembly routine:  https://github.com/fdrmrc/AggloDeal/blob/8387ca9430109bf4315536603f9e1d0467cf18c1/examples/poisson.cc#L185-L408 which should be much cleaner. The only `if`s are:
- the usual one that checks it a face of a cell is a boundary face or not
- the one checking that each face is seen only once during the process

but the user is not exposed to internal details of the agglomeration anymore. The 4 local matrices to compute face contributions are still there. 

@luca-heltai

**Comment by luca-heltai** [comment]
This looks perfect to me. Except for a couple of very minor details, I think we are very close to a definitive API. The only missing thing to make this exceptionally perfect is the FEInterfaceValues interface, which can be done in a followup change.

Bravo!
